src/test/java/io/strimzi/kafka/bridge/mqtt/MqttBridgetIT.java

```diff
@@ -0,0 +1,252 @@
+import org.slf4j.LoggerFactory;
+
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.UUID;
+import java.util.List;
+import java.util.ArrayList;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Integration test for MQTT bridge
+ */
+public class MqttBridgetIT {
+
+    private static final Logger logger = LoggerFactory.getLogger(MqttBridgetIT.class);
+    private static final String MQTT_SERVER_HOST = "localhost";
+    private static final int MQTT_SERVER_PORT = 1883;
+    private static final String MQTT_SERVER_URI = "tcp://" + MQTT_SERVER_HOST + ":" + MQTT_SERVER_PORT;
+    private static final String KAFKA_TOPIC = "devices_bluetooth_data";
+    private static KafkaConsumer<String, String> kafkaConsumerClient = null;
+    private static MqttServer mqttBridge;
+    private static Thread prepareServerThread;
+    private static StrimziKafkaContainer kafkaContainer;
+    private String mqttClientId;
+
+
+    /**
+     * Start the kafka Cluster
+     * Start the MQTT bridge before all tests
+     */
+    @BeforeAll
+    public static void beforeAll() throws InterruptedException {
+        final Map<String, Object> config = new HashMap<>();
+        String kafkaBootstrapServers = "localhost:9092";
+        try {
+            kafkaContainer = new StrimziKafkaContainer();
+            kafkaContainer.waitForRunning();
+            kafkaContainer.start();
+            kafkaBootstrapServers = kafkaContainer.getBootstrapServers();
+        } catch (Exception e) {
+            logger.error("Exception occurred", e);
+        }
+
+        // prepare the configuration for the bridge
+        config.put(KafkaConfig.KAFKA_CONFIG_PREFIX + ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers);
+        config.put(MqttConfig.MQTT_HOST, MQTT_SERVER_HOST);
+        config.put(MqttConfig.MQTT_PORT, MQTT_SERVER_PORT);
+        config.put(BridgeConfig.BRIDGE_ID, "my-bridge");
+
+        // instantiate the kafka consumer client
+        kafkaConsumerClient = new KafkaConsumer<>(
+                ImmutableMap.of(
+                        ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers,
+                        ConsumerConfig.GROUP_ID_CONFIG, "gid-" + UUID.randomUUID(),
+                        ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OffsetResetStrategy.EARLIEST.name().toLowerCase(Locale.ROOT)
+                ),
+                new StringDeserializer(),
+                new StringDeserializer()
+        );
+
+        // consumer client subscribes to the kafka topic
+        kafkaConsumerClient.subscribe(Collections.singletonList(KAFKA_TOPIC));
+
+        // start the bridge server in a separate thread
+        prepareServerThread = new Thread(() -> {
```

Comment:
Before this, I was using a CountDownLatch to coordinate the thread. That's why I called this thread prepareServerThread, because it would have the server execution.
But it was not working as expected, so removed the CountDownLatch and forgot to rename it according.